src/aof.c

```diff
@@ -874,7 +874,7 @@ int loadAppendOnlyFile(char *filename) {
             serverLog(LL_WARNING,
                 "Unknown command '%s' reading the append only file",
                 (char*)argv[0]->ptr);
-            exit(1);
+            return AOF_CRIT;
```

Comment:
now that we handle that error (and don't necessarily lead to an `exit`), we need to cleanup after us.
for instance, free the `argv` array.

at first it looked as if the pre-existing (possibly successful) terminations in this function, for instance, the `goto readerr` also didn't release the `argv` array, but then i noticed it is attached to the client.
so i suppose all there is to do is release the fake client before returning.

on the other hand, the `stopLoading` is also probably missing, so maybe it's best to set some error type variable, and `goto` some common cleanup section at the bottom.

Reply:
- I added a test for AOF with unknown command.
- Added freeFakeClientArgv that still needed
- New label of cleanup added